### PR TITLE
[TIMOB-24503] Android: Implement notification quick actions

### PR DESCRIPTION
**Pull request by garymathews**
###### Requires support library `26.0.0` https://github.com/appcelerator/titanium_mobile/pull/9274

- Implement notification quick actions
- Expose support library `26.0.0` feature `setGroupAlertBehavior()`

###### TEST CASE
```JS
var win = Ti.UI.createWindow({title: 'TIMOB-24503', layout: 'vertical'}),
    btn = Ti.UI.createButton({title: 'CREATE NOTIFICATION', height: 50, width: Ti.UI.FILL}),
    id = 100;

function createNotification() {
        var n = Ti.Android.createNotification({
                icon: Ti.Android.R.drawable.ic_dialog_info,
                contentTitle: 'APPCELERATOR',
                contentText : 'Open Appcelerator web-page.'
            }),
            p = Ti.Android.createPendingIntent({
                intent: Ti.Android.createIntent({
                    action: Ti.Android.ACTION_VIEW,
                    data: 'https://www.appcelerator.com'
                })
            });
        n.addAction(Ti.Android.R.drawable.ic_dialog_email, "OPEN", p);
        Ti.Android.NotificationManager.notify(id, n);
}

btn.addEventListener('click', function() {
    createNotification();
});

win.add([btn]);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-24503)

### CODE REVIEW

File: apidoc/Titanium/Android/Notification.yml [hunk 1]
```diff
@@ -154,6 +154,22 @@ methods:
         summary: Allows to enable the indeterminate mode.
         type: Boolean
 
+  - name: addAction
+    summary: Add an action button to the notification
+    since: "6.2.0"
+    parameters:
+      - name: icon
+        summary: Icon of the action button as URL or resource ID
+        type: [Number, String]
```
Comment (jquick-axway) [inline apidoc/Titanium/Android/Notification.yml]: I think the icons have to use a holo dark theme on Android 4.x and a light material theme on Android 5.0 and above. Which means developers need to provide versioned drawable resources (ie: "drawable-mdpi-v21" for material). Or perhaps those details should go into the notification guide?

File: android/modules/android/src/java/ti/modules/titanium/android/notificationmanager/NotificationProxy.java [hunk 2]
```diff
@@ -371,6 +366,24 @@ public void setProgress(int max, int progress, boolean indeterminate)
...
+		} else {
+			String iconUrl = TiConvert.toString(icon);
+			if (iconUrl == null) {
+				Log.e(TAG, "Url is null");
+				return;
+			}
+			String iconFullUrl = resolveUrl(null, iconUrl);
+			iconId = TiUIHelper.getResourceId(iconFullUrl);
+		}
+		notificationBuilder.addAction(iconId, title, pendingIntent.getPendingIntent());
```
Comment (jquick-axway) [inline android/modules/android/src/java/ti/modules/titanium/android/notificationmanager/NotificationProxy.java]: Perhaps we should do the following if title is null. (I'm thinking icon-only action buttons.)
```java
if (title == null) {
    title = "";
}
```

Comment (jquick-axway) [inline android/modules/android/src/java/ti/modules/titanium/android/notificationmanager/NotificationProxy.java]: And if "pendingIntent" is null, we should log an error or throw an exception indicating that this argument cannot be null.

**Review by jquick-axway — CHANGES_REQUESTED** [review]

**Comment by garymathews** [comment]
@jquick-axway Updated PR, `title` can be `null` for icon only buttons

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Comment by lokeshchdhry** [comment]
FR Passed.

1. Checked with code above.
2. Checked with `data: 'http://maps.google.com/maps?q=loc:37.3906238,-122.0499305,19'` which opens up the maps app.
3. Checked with :
```
intent : Ti.Android.createIntent({
	action: Ti.Android.ACTION_VIEW,
	className: 'com.app.zxyz.ZxyzActivity',
	packageName: 'com.app.zxyx
})
```
4. Checked with `null` argument's for `title` & pending intent`.

Studio Ver:     4.9.1.201707200100
SDK Ver:       7.0.0 local build
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.3
Appc NPM:       4.2.9
Appc CLI:       6.2.3
Ti CLI Ver:     5.0.14
Alloy Ver:      1.9.13
Node Ver:       6.10.1
Java Ver:       1.8.0_101
Devices:        ⇨ google Nexus 5 --- Android 6.0.1
                ⇨ google Pixel --- Android 7.1.1

**Pull request merged**